tools/stress-cluster/services/Stress.Watcher/src/PodEventHandler.cs

```diff
@@ -215,5 +228,61 @@ public bool ShouldStartPodChaos(WatchEventType type, V1Pod pod)
 
             return true;
         }
+
+        public async Task DeleteResources(V1Pod pod)
+        {
+            if (!ShouldDeleteResources(pod))
+            {
+                Logger.Debug("Skipping pod.");
+                return;
+            }
+
+            var rgName = GetResourceGroupName(pod);
+
+            if (rgName == "")
+            {
+                Logger.Error("Failed to find resource group name.");
```

Comment:
This log statement could also be a little misleading because it isn't necessarily a failure condition. I think some of the logic from `GetResourceGroupName` should be moved into the `ShouldDeleteResources` check instead.